File: v2-enterprise/mobile-app/src/types/installment.ts
export interface Installment {
  id: string;
  installment_number: number;
  due_date: string;
  amount: number;
  paid_amount: number;
  remaining: number;
  status: 'Paid' | 'Pending' | 'PartiallyPaid' | 'Overdue';
  payment_date?: string;
  contract_number: string;
}

export interface PaymentHistory {
  id: string;
  reference_number: string;
  amount: number;
  payment_date: string;
  method: string;
  description: string;
}

File: v2-enterprise/src/Domain/Entities/Installment.cs
using System;
using CarShowroomManagementV2.Domain.Common;

namespace CarShowroomManagementV2.Domain.Entities
{
    public class Installment : AuditableEntity
    {
        public Guid InstallmentPlanId { get; set; }
        public int InstallmentNumber { get; set; } // رقم الدفعة (مثال: 1, 2...)
        public DateTime DueDate { get; set; } // تاريخ استحقاق الدفعة
        public decimal Amount { get; set; } // قيمة القسط المستحقة
        public decimal PaidAmount { get; set; } = 0; // المبلغ الذي تم سداده فعلياً
        public string Status { get; set; } = "Pending"; // Pending, Paid, PartiallyPaid, Overdue, Cancelled
        public DateTime? PaymentDate { get; set; } // تاريخ السداد المالي الفعلي
        public bool IsProfitRecognized { get; set; } = false;

        // علاقة التنقل
        public virtual InstallmentPlan? InstallmentPlan { get; set; }
    }
}
